### DISCUSSION TITLE
OpenWrt 安装 MosDNS v5 超傻瓜方式

### DISCUSSION THREAD

**Discussion by qingtian110** [IrineSistiana/mosdns#455] [Ideas]
### 来源：https://github.com/sbwml/luci-app-mosdns

# 🎁 MosDNS v5 is Ready! 🎉
## 正在使用v4 版本的童鞋，可以用安装脚本无障碍升级到 v5！

-------------------------

⚠注意⚠

曾经安装过其他版本的 MosDNS LuCI 或使用的固件自带了 MosDNS，安装后可能无法正常运行，你需要先将其它来源的版本卸载并清除数据残留后再装才能使用。

解决方法：

1、卸载包含 **mosdns** 字符的所有软件包
2、删除数据 `rm -rf /etc/config/mosdns /etc/mosdns`
3、重新安装 mosdns

-------------------------

## How to install prebuilt packages (OpenWrt 18-22)

- Login OpenWrt terminal (SSH)

- Install `curl` package
  ```shell
  opkg update
  opkg install curl
  ```

- Execute install script (Multi-architecture support)
  ```shell
  sh -c "$(curl -ksS https://raw.githubusercontent.com/sbwml/luci-app-mosdns/v5/install.sh)"
  ```

----------------------------------

# 插件截图

### 普通用户的默认配置（一般来讲已经非常够用）

- 默认配置
![1](https://user-images.githubusercontent.com/87856890/229039926-eee7d563-ccff-483f-a929-c421c34344f2.png)
![2](https://user-images.githubusercontent.com/87856890/229039617-fa1a3701-3998-4ebd-82a7-718d123cfbde.png)
![3](https://user-images.githubusercontent.com/87856890/229039633-c0229de9-eae6-403e-9981-7f12afa84cba.png)

- 规则列表
 白名单：**加入的域名始终允许使用 “本地 DNS” 进行解析，且优先级最高（每个域名一行，支持域名匹配规则）**
 黑名单：**加入的域名将屏蔽 DNS 解析（每个域名一行，支持域名匹配规则）**
 灰名单：**加入的域名始终使用 “远程 DNS” 进行解析（每个域名一行，支持域名匹配规则）**
DDNS 域名：**加入的域名始终使用 “本地 DNS” 进行解析，并且强制 TTL 5 秒，解析结果不会进入缓存（每个域名一行，支持域名匹配规则）**
 Hosts：**自定义 Hosts 重写，如：baidu.com 10.0.0.1（每个规则一行）**
 重定向：**重定向请求的域名。请求域名 A，但返回域名 B 的记录，如：baidu.com qq.com（每个规则一行）**
 PTR 黑名单：**加入的域名将阻止 PTR 请求（每个域名一行，支持域名匹配规则）**

![4](https://user-images.githubusercontent.com/87856890/229039665-7b09196b-fcda-42d1-b9bd-ea2aeea10028.png)

- 数据库更新（geodata，它提供广告列表、大陆域名、ip归属）
![22](https://user-images.githubusercontent.com/87856890/182366257-d73f011a-9960-4b10-9d09-0a939821b726.png)

- 日志文件
![33](https://user-images.githubusercontent.com/87856890/182366272-c5af34a4-7e02-49ac-b627-8779f021984b.png)

### 进阶用户的 DIY 配置

![11](https://user-images.githubusercontent.com/87856890/229040149-034b7083-0c77-4a67-a676-de3fc7f54f3e.png)
![22](https://user-images.githubusercontent.com/87856890/229040169-1c3dbc80-f0c6-4e44-aa98-cda5e4047e5e.png)

**Comment by hoicau** [comment]


**Reply by cicai1631** [reply]
root@OpenWrt:~# sh -c "$(curl -sS https://raw.githubusercontent.com/sbwml/luci-app-mosdns/master/install.sh)"
Error loading shared library libmbedtls.so.14: No such file or directory (needed by /usr/bin/curl)
Error loading shared library libmbedx509.so.1: No such file or directory (needed by /usr/bin/curl)
Error loading shared library libmbedcrypto.so.7: No such file or directory (needed by /usr/bin/curl)
Error relocating /usr/bin/curl: curl_easy_header: symbol not found
Error relocating /usr/bin/curl: curl_easy_nextheader: symbol not found

**Reply by qingtian110** [reply]
> root@OpenWrt:~# sh -c "$(curl -sS https://raw.githubusercontent.com/sbwml/luci-app-mosdns/master/install.sh)" Error loading shared library libmbedtls.so.14: No such file or directory (needed by /usr/bin/curl) Error loading shared library libmbedx509.so.1: No such file or directory (needed by /usr/bin/curl) Error loading shared library libmbedcrypto.so.7: No such file or directory (needed by /usr/bin/curl) Error relocating /usr/bin/curl: curl_easy_header: symbol not found Error relocating /usr/bin/curl: curl_easy_nextheader: symbol not found

你的curl 炸了，盲猜是lede固件吧

**Comment by coolzhi** [comment]
求个自定义配置，用默认的有点问题个别网站

**Comment by hyperlook** [comment]
牛逼，顶你下

**Comment by birdfly** [comment]
sh -c "$(curl -ksS https://raw.githubusercontent.com/sbwml/luci-app-mosdns/master/install.sh)"
curl: (48) Error

这个问题如何解决

**Reply by wslpdgg** [reply]
opkg update  libcurl ,libcurl和curl版本要一致

**Reply by susan039157** [reply]
我也是这种情况  要怎么解决呢

**Reply by susan039157** [reply]
> opkg update libcurl ,libcurl和curl版本要一致

感谢  解决了

**Comment by zqp561022** [comment]
这个NB，对于没有编译mosdns的固件是个福音

**Comment by wuwentao** [comment]
op旁路由模式，直接禁用dnsmasq了（改成了其他不用的端口），然后把mosdns直接监听53端口，然后登录op执行报错，只好手工一行一行执行，看看哪里出错，发现取不到ip
ip_info=$(curl -sk https://ip.cooluc.com)
但是在op上ping这个域名是通的，放弃治疗了，
直接看sh源码，手工去下载了几个release包，手工opkg install来升级成功了。
自带的自定义配置好像也够用了，还没仔细看

**Comment by anson0459** [comment]
Error! download https://github.com/sbwml/luci-app-mosdns/releases/download/v5.1.3-4/mosdns_5.1.3-4_x86_64.ipk failed.
下载不下来了

**Comment by jack-004** [comment]
安装出面以下错误 Error relocating /usr/lib/libcurl.so.4: nghttp2_option_set_no_rfc9113_leading_and_trailing_ws_validation: symbol not found

**Reply by sms1007** [reply]
Error relocating /usr/bin/curl: curl_global_trace: symbol not found
我是这样的

**Reply by wangn9900** [reply]
opkg update; opkg install libcurl4

**Comment by hww-png** [comment]
我这边安装也报错了，提示sh ：can't open h1 :no such file
![image](https://github.com/IrineSistiana/mosdns/assets/77440291/ddb3933e-2da0-4fe2-9897-716549403c9a)

**Comment by TeddyXiong79** [comment]
hello，问问已经安装好了5.1.3-10，怎么升级哦？ 是不是直接运行安装时的脚本哦》》》》
sh -c "$(curl -ksS https://raw.githubusercontent.com/sbwml/luci-app-mosdns/v5/install.sh)"

**Comment by lovestudygithub** [comment]
这个依赖一个什么组件的（忘记了），如果要用5.3.1的版本，需要卸载掉这个组件的12版本，升级到13和14的版本，不过这样会掉三个软件，helloword、bypass、passwall（passwall2倒是没事）和ShadowSocksR Plus+，我用openclash，所以掉了也无所谓。我用的esir的x86固件。

**Comment by mangfei** [comment]
人才部部长~~~搞定

**Comment by infopg** [comment]
root@BleachWrt:~# sh -c "$(curl -ksS https://raw.githubusercontent.com/sbwml/luci-app-mosdns/v5/install.sh)"
Error loading shared library libmbedtls.so.21: No such file or directory (needed by /usr/bin/curl)
Error loading shared library libmbedx509.so.7: No such file or directory (needed by /usr/bin/curl)
Error loading shared library libmbedcrypto.so.16: No such file or directory (needed by /usr/bin/curl)

**Comment by uyonger** [comment]
如果安装以后，一直无法启动，，可以试试 把9091 端口 空出来，，我原来把外部端口9091 给下载使用，，，导致一直无法启动，，，后面改成9092，，空出来9091 就可以了

**Comment by alal167** [comment]
请问immwrt安装不上是怎么回事

**Reply by ZacharyRen77** [reply]
我也是这种情况

**Reply by d4rkblue** [reply]
同样的情况

**Reply by d4rkblue** [reply]
解决了。OP空间不够，扩容后通过命令行安装成功。

**Comment by kdatamr** [comment]
安装后服务菜单看不到mosdns

**Reply by qingtian110** [reply]
> 安装后服务菜单看不到mosdns

现在 luci 作者用 js 重写了。。像lede 、openwrt 18 这些超级老的 luci 不支持了。

**Reply by Foch0x97** [reply]
+1

**Comment by kidxiang** [comment]
怎么安装5.3.1版本呢？

**Reply by Bornres** [reply]
同问

**Comment by OiCkilL** [comment]
新版安装后选项少了好多

**Comment by xiaoyaozhilv** [comment]
进阶配置也说一说啊，大佬

**Comment by irvinow** [comment]
安装完没有lib.sh和set.sh文件，有大神知道原因么

**Comment by thinker226** [comment]
可以，安装之后就是最新的
